src/stream/quic_lb/ngx_stream_quic_lb_retry_service.c

```diff
@@ -364,20 +398,36 @@ ngx_stream_quic_lb_validate_timestamp(uint64_t expire_time)
  * ++++++++++++++++++++++++++++++++++++++
  */
 static ngx_int_t
-ngx_stream_quic_lb_gen_share_state_plain_token_body(ngx_connection_t *c, 
-    retry_token_enc_info_t *enc_info, ngx_str_t *cids, u_char *buf, 
+ngx_stream_quic_lb_gen_share_state_plain_token_body(ngx_connection_t *c,
+    retry_token_enc_info_t *enc_info, ngx_str_t *cids, u_char *buf,
     ngx_int_t buf_len, size_t *out_len)
 {
     ngx_int_t                           token_body_len = 0;
-    size_t                              len;
     ngx_quic_lb_retry_token_body_t      token_body;
     ngx_str_t                           odcid, rscid;
+#ifdef NGX_QUIC_DEBUG_CRYPTO
+    u_char                             *debug_port_p;
+    u_char                             *debug_expire_time_p;
+#endif
 
     odcid = cids[0];
     rscid = cids[2];
 
-    token_body.odcid_len = odcid.len;
-    token_body.rscid_len = rscid.len;
+    if (odcid.len == 0) {
```

Comment:
the cid rule described in  section-7.3.3  is ODCID and RSCID in token not dcid in retry packet, so when do token validation this rule should check.

But when generating retry packet, dcid's len should obey the rule in PR:
https://github.com/alipay/quic-lb/pull/9